modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/upgrade/test/SXPBlueprintUpgradeProcessTest.java

```diff
@@ -0,0 +1,480 @@
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.search.experiences.model.SXPBlueprint;
+import com.liferay.search.experiences.model.SXPElement;
+import com.liferay.search.experiences.rest.dto.v1_0.Configuration;
+import com.liferay.search.experiences.rest.dto.v1_0.GeneralConfiguration;
+import com.liferay.search.experiences.rest.dto.v1_0.util.ConfigurationUtil;
+import com.liferay.search.experiences.service.SXPBlueprintLocalService;
+import com.liferay.search.experiences.service.SXPElementLocalService;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Almir Ferreira
+ */
+@RunWith(Arquillian.class)
+public class SXPBlueprintUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		_user = TestPropsValues.getUser();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group, _user.getUserId());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		Iterator<Long> sxpBlueprintIdsIterator = _sxpBlueprintIds.iterator();
+
+		while (sxpBlueprintIdsIterator.hasNext()) {
+			_sxpBlueprintLocalService.deleteSXPBlueprint(
+				sxpBlueprintIdsIterator.next());
+
+			sxpBlueprintIdsIterator.remove();
+		}
+
+		Iterator<Long> sxpElementIdsIterator = _sxpElementIds.iterator();
+
+		while (sxpElementIdsIterator.hasNext()) {
+			_sxpElementLocalService.deleteSXPElement(
+				sxpElementIdsIterator.next());
+
+			sxpElementIdsIterator.remove();
+		}
+	}
+
+	@Test
+	public void testUpgradeEmptySXPBlueprintAndSXPElement() throws Exception {
+		SXPBlueprint sxpBlueprint = _addSXPBlueprint(null, null, null);
+		SXPElement sxpElement = _addSXPElement(null);
+
+		_runAllUpgrades();
+
+		_assertSXPBlueprint("[]", sxpBlueprint);
+		_assertSXPElement("", sxpElement);
+	}
+
+	@Test
+	public void testUpgradePortletPreferenceValue() throws Exception {
+		SXPBlueprint sxpBlueprint = _addSXPBlueprint(
+			_sxpBlueprintConfigurationJSONObject.toString(), null, null);
+
+		Long sxpBlueprintId = sxpBlueprint.getSXPBlueprintId();
+
+		PortletPreferenceValue searchBarSuggestions =
+			_addPortletPreferenceValue(
+				"suggestionsContributorConfigurations", _SEARCH_BAR_PORTLET_ID,
+				JSONUtil.put(
```

Comment:
extract this json and then use .put to change value of "attributes" when assert instead of repeat it in line 164